cloudinit/config/cc_apt_configure.py

```diff
@@ -680,7 +680,13 @@ def generate_sources_list(cfg, release, mirrors, cloud):
             )
             aptsrc_file = apt_sources_list
     disabled = disable_suites(cfg.get("disable_suites"), rendered, release)
+    disable_apt_sources_list = False
+    if aptsrc_file == apt_sources_deb822 and os.path.exists(apt_sources_list):
+        disable_apt_sources_list = True
     util.write_file(aptsrc_file, disabled, mode=0o644)
+    if disable_apt_sources_list:
+        LOG.debug("Disabling %s", apt_sources_list)
+        util.rename(apt_sources_list, f"{apt_sources_list}.old")
```

Comment:
I think we want to rename this `.disabled` instead of `.old` as a best practice for preserved and ignored apt sources files as we see apt ignores these files by default `
```
$ apt-config dump | grep disable
Dir::Ignore-Files-Silently:: "\.disabled$";
```

I think we may also want to add a header line `# disabled by cloud-init` to this file to leave a breadcrumb for why this rename happened